Utils/trigger_contribution_build.py

```diff
@@ -0,0 +1,114 @@
+from gitlab.v4.objects.branches import ProjectBranch
+from gitlab.v4.objects.projects import Project
+
+urllib3.disable_warnings()
+
+CONTRIBUTION_PRS_QUERY = 'is:pull-request state:open label:ready-for-instance-test label:Contribution label:"External PR"'
+GITLAB_PROJECT_ID = os.getenv("CI_PROJECT_ID") or 1061
+GITLAB_SERVER_URL = os.getenv("CI_SERVER_URL", "https://gitlab.xdr.pan.local")  # disable-secrets-detection
+GITHUB_TRIGGER_BUILD_LABEL = "ready-for-instance-test"
+
+MESSAGES = namedtuple(
+    "MESSAGES",
+    ["build_request_accepted", "build_triggered", "build_trigger_failed"],
+)
+COMMENT_MESSAGES = MESSAGES(
+    "For the Reviewer: Trigger build request has been accepted for this contribution PR.",
+    "For the Reviewer: Successfully created a pipeline in GitLab with url: {url}",
+    "For the Reviewer: Could not create a pipeline in GitLab for branch name: {branch}. \
+    The branch was not found in the GitLab project.",
+)
+
+
+def arguments_handler() -> argparse.Namespace:
+    """Validates and parses script arguments.
+
+    Returns:
+       Namespace: Parsed arguments object.
+
+    """
+    parser = argparse.ArgumentParser(description="Trigger contribution build.")
+    parser.add_argument("--github-token", help="Github API token")
+    parser.add_argument("--gitlab-api-token", help="Gitlab API token")
+    parser.add_argument("--gitlab-trigger-token", help='Gitlab trigger token')
+    return parser.parse_args()
+
+
+def cancel_active_pipelines(gitlab_project: Project, branch: ProjectBranch) -> None:
+    """Find and cancel all currently active pipelines for a given GitLab branch.
+
+    Args:
+        gitlab_project (Project): GitLab Project object.
+        branch (ProjectBranch): Gitlab branch object.
+    """
+    pipelines = gitlab_project.pipelines.list(ref=branch.name, status="running")
+    if pipelines:
+        for pipeline in pipelines:
+            print(f"Canceling active pipeline: {pipeline.id}")
+            pipeline.cancel()
+
+
+def handle_contribution_prs(args, github_issues: PaginatedList[Issue], gitlab_project: Project) -> None:
+    """Given a list of github issues (PRs), create new pipelines for their mirrored GitLab branches.
+    Older active pipelines for each branch will be canceled.
+
+    Args:
+        args (Namespace): Script arguments.
+        github_issues (PaginatedList[Issue]): List of GitHub PRs returned from the contribution PRs query.
+        gitlab_project (Project): GitLab project object.
+    """
+    for issue in github_issues:
+        issue.create_comment(COMMENT_MESSAGES.build_request_accepted)
+        pull_request = issue.as_pull_request()
+        github_branch_name = pull_request.head.ref
+
+        # get the corresponding GitLab branch object corresponding to the GitHub branch
+        if branch := gitlab_project.branches.get(github_branch_name):
+            print(f'--- Handling branch: {branch.name}. ---')
+
+            cancel_active_pipelines(gitlab_project, branch)
```

Comment:
Lets check if it is possible to handle errors gracefully as a comment.

Reply:
try-except has been added where necessary.